lib/timeline/timeline.dart

```diff
@@ -60,13 +80,17 @@ class TimelineScreen extends Screen {
       ..disabled = true
       ..click(_resumeRecording);
 
-    final PButton trackWidgetBuildsButton = new PButton('Track widget builds')
-      ..small();
-    final PButton perfOverlayButton = new PButton('Performance overlay')
-      ..small();
-    final PButton repaintRainbowButton = new PButton('Repaint rainbow')
-      ..small();
-    final PButton debugDrawButton = new PButton('Debug draw')..small();
+    PButton createButton(String text, bool shouldDisableForProfileMode) {
+      return new PButton(text)
+        ..small()
+        ..disabled = shouldDisableForProfileMode && isProfileMode
+        ..tooltip = isProfileMode ? 'Unavailable on a profile build.' : '';
+    }
+
+    final PButton trackWidgetBuildsButton = createButton('Track widget builds', true);
+    final PButton perfOverlayButton = createButton('Performance overlay', false);
```

Comment:
these should be turned on and off by similar logic to VMServiceManager in the Java app instead of triggering based on whether we are in debug or profile mode. That is more robust as potentially flutter will support some or all of these options in profile mode.